components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -33,6 +35,27 @@ import { Store } from '../../common/async/types'
 import { HardwareWalletAccount } from 'components/brave_wallet_ui/common/hardware/types'
 import { GetTokenParam } from '../../utils/api-utils'
 
+function encodeKeyToHex (key: string): string {
+  return Buffer.from(key, 'base64').toString('hex')
+}
+
+function switchEndianness (hexString: string): string | false {
+  const regex = hexString.match(/.{2}/g)
+  if (!regex) {
+    return false
+  }
+  return regex.reverse().join('')
+}
+
+function extractPublicKeyForBLS (privateKey: string): string {
```

Comment:
would be good to have some unit tests for this

Reply:
added it in next PR where I moved this function to keyring module https://github.com/brave/brave-core/pull/11396/files#diff-a55a419b38e50bf2369a26ee7953fe2607905d62fdd10d28bc611ba30e6e1fadR33

Reply:
this doesn't seem like something we should be doing in JS to me. I think it would be better to call out to native so JS is strictly focused on UI interactions and does not contain any logic for keys, endianness, etc...

Reply:
Per discussion with @bridiver on slack, I think we should do this issue in this PR
https://github.com/brave/brave-browser/issues/20024
since we are introducing new dependency in js side

Reply:
We have another issue for this https://github.com/brave/brave-browser/issues/20024 with ongoing work to add rust lib for BLS to native side. It will be too big to mix it in one PR. Additionally we need to solve some questions with Filecoin team to integrate it correctly. Since the security review has been closed for this lib let's use this deps testing/development purposes until I will solve issues with connecting rust lib in parallel

Reply:
sounds good to me